zipline/history/history.py

```diff
@@ -266,6 +270,9 @@ def __init__(self, bar_count, frequency, field, ffill,
                  data_frequency='daily'):
 
         # Number of bars to look back.
+        if not isinstance(bar_count, int):
+            raise ValueError('bar_count must be an int')
```

Comment:
This should reference the actual received type so the user can do something about it.